src/Common/src/Interop/Windows/Winsock/Interop.WinsockAsync.cs

```diff
@@ -161,14 +165,29 @@ internal struct NetworkEvents
         }
 
         [StructLayout(LayoutKind.Sequential)]
-        internal struct WSAMsg
+        internal unsafe struct WSAMsg
         {
-            internal IntPtr socketAddress;
+            internal byte* socketAddress;
             internal uint addressLength;
-            internal IntPtr buffers;
+            internal WSABuffer* buffers;
             internal uint count;
             internal WSABuffer controlBuffer;
             internal SocketFlags flags;
+            internal int pad;
+
+            // NOTE: Even though Win32 docs say the flags arg is a DWORD, and thus 4 bytes long, it seems to actually write 4 bytes past this.
+            // The "pad" field above deals with this.
```

Comment:
This seems like a padding issue for 32-bit vs 64-bit and should be implicitly handled by the runtime (it will also be brokn on 32-bit, as the struct will be 32-bytes, rather than 28-bytes).

On 32-bit:
* sizeof(WSAMSG) = 28-bytes
* offsetof(name) = 0
* offsetof(namelen) = 4
* offsetof(lpBuffers) = 8
* offsetof(dwBufferCount) = 12
* offsetof(Control.len) = 16
* offsetof(Control.buf) = 16
* offsetof(dwFlags) = 24

On 64-bit:
* sizeof(WSAMSG) = 56-bytes
* offsetof(name) = 0
* offsetof(namelen) = 8
* offsetof(lpBuffers) = 16
* offsetof(dwBufferCount) = 24
* offsetof(Control.len) = 32
* offsetof(Control.buf) = 40
* offsetof(dwFlags) = 48